
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>文献分享-metagenome binning - Blog{Rachel|BENM}</title>
  <meta name="author" content="Wu Xiaomeng(Rachel)">

  
  <meta name="description" content="文章来源 Albertsen M, Hugenholtz P, Skarshewski A, et al. Genome sequences of rare, uncultured bacteria obtained by differential coverage binning of &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://rachelwxm.github.io/blog/2014/09/18/wen-xian-fen-xiang-genome-sequences-of-rare/">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="/javascripts/ender.js"></script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <link href="/atom.xml" rel="alternate" title="Blog{Rachel|BENM}" type="application/atom+xml">
  <link href='http://fonts.googleapis.com/css?family=Lato:400,700,400italic,700italic' rel='stylesheet' type='text/css'>
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href="http://fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
<link href="http://fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">


  

</head>

<body   >
  <header role="banner"><hgroup>
  <div id="logo">
 	<div id="logoText"></div>
  	<div id="logoLeft"></div>
  	<div id="logoText"></div>
  	<div id="logoRight"></div>
  	<div class="clear"></div>
  </div>
  <h1><a href="/">Blog{Rachel|BENM}</a></h1>
  
    <h2>A blog for bioinformatics learning</h2>
  
  <div class="clear"></div>
</hgroup>

</header>
  <nav role="navigation"><ul class="subscription" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS">RSS</a></li>
  
</ul>
  
<form action="http://google.com.hk/search" method="get">
  <fieldset role="search">
    <input type="hidden" name="q" value="site:rachelwxm.github.io" />
    <input class="search" type="text" name="q" results="0" placeholder="Search"/>
  </fieldset>
</form>
  
<ul class="main-navigation">
  <li><a href="/">Blog</a></li>
  <li><a href="/blog/archives">Archives</a></li>
  <li><a href="/about">About Me</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div>
<article class="hentry" role="article">
  
  <header>
    
      
        <h1 class="entry-title">文献分享-metagenome Binning</h1>
      
    
    
      <p class="meta">
        








  


<time datetime="2014-09-18T12:06:07+08:00" pubdate data-updated="true">Sep 18<span>th</span>, 2014</time>
        
      </p>
    
  </header>


<div class="entry-content"><h4>文章来源</h4>

<p><a href="http://www.nature.com/nbt/journal/v31/n6/abs/nbt.2579.html">Albertsen M, Hugenholtz P, Skarshewski A, et al. Genome sequences of rare, uncultured bacteria obtained by differential coverage binning of multiple metagenomes[J]. Nature biotechnology, 2013, 31(6): 533-538.</a><br/>
基于宏基因组覆盖度差异获得罕见的、非培养的细菌基因组序列</p>

<h4>背景和意义</h4>

<ul>
<li>意义： 从非培养的宏基因组中分离出单个细菌基因组</li>
<li>宏基因组binning方法

<ul>
<li>基于序列相似性比对</li>
<li>基于物种分类标记物基因</li>
<li>基于序列特征（GC）</li>
<li>基于关联性特征，结合机器学习</li>
</ul>
</li>
<li>局限性

<ul>
<li>数据库中缺乏相应的序列</li>
<li>不是所有的宏基因组 DNA 都包含标记物基因，标记物基因种间水平转移</li>
<li>序列短，序列覆盖度低</li>
<li>计算功能、算法要求高</li>
</ul>
</li>
</ul>


<!--more-->


<p></p>

<h4>样品来源</h4>

<p>丹麦某污水处理厂的活性污泥反应器（activated sludge bioreactor）</p>

<h4>数据来源</h4>

<p>两种不同的DNA提取方法：</p>

<ul>
<li>热酚处理（HP+：29G）-破坏细菌细胞壁，降低部分菌株的含量</li>
<li>非热酚处理（HP-：57G）</li>
</ul>


<h4>研究方法</h4>

<p>[fig1_flow<em>]
<img alt='Fig 1. Overview of the pipeline to obtain high-quality population genomes from multiple deep metagenomes using differential coverage as the primary binning method, illustrated using the population genome TM7-AAU-ii.'  src='/images/POST/MetaBin/meta_bin_flow.png'><br/>
Step1：采样（包含多种微生物）；<br/>
Step2：使用两种方法提取宏基因组DNA，这两种方式会导致不同物种DNA在宏基因组DNA中相对丰度的差异；<br/>
Step3：使用Hiseq2000测序平台，PE150策略进行测序；<br/>
Step4：对原始的测序数据（Raw Data）进行质控、</em>de novo_组装，因为HP+处理后的样本测序量低而且测出来的reads有物种偏向性，最后只用了HP-处理的数据组装的scaffold做后续分析；</p>

<blockquote><p>However, because fewer reads were sequenced and there was a skewed spe- cies abundance toward few organisms in the HP+ metagenome, only the HP− metagenome scaffolds were used for extraction of full genomes</p></blockquote>

<p>Step5：Hp+和HP-的reads与scaffold比对（mapping），计算每个scaffold在两种处理下的相对丰度；<br/>
Step6：对每个分理处的bin进行基于四核苷酸频率的PCA分析；<br/>
Step7：统计每个scaffold的GC含量；<br/>
Step8：标记基因鉴定：用HMMER3软件预测scaffold的ORF区域，再与RefSeq数据库比对注释；</p>

<p>Step9：将scaffold按照它们在两种处理中的相对丰度聚类（binning），如图2所示，图中横坐标是各scaffold在HP-处理中的相对丰度，纵坐标是在HP+处理中的相对丰度；聚类原理是：当在菌株级别对样品做处理时，来自同一个物种的DNA会整体表现降低，相对的没有受到热酚影响的物种的DNA丰度会整体表现提高；<br/>
<img alt='Fig 2. Sequence composition–independent binning of metagenome scaffolds from the lab-scale bioreactor using differential coverage (HP+, HP−).' src='/images/POST/MetaBin/composition–independent binning.png'><br/>
Step10：利用PE reads将已分类的bin中的序列与多拷贝区域连接起来，再用Cytoscape软件画出连接图；<br/>
Step11：将所有mapping到一个bin的reads重新组装；<br/>
Step12：评估组装出“单菌”的基因组DNA完整性，用Circle画圈图。</p>

<h4>研究结果</h4>

<p>从宏基因组测序结果中组装出序列量小并且无参的基因组，组装得到31个细菌基因组（包括相对低丰度&lt;1%的），其中12个已经或接近得到完整基因组信息，12个中的4个属于TM7门。</p>

<h4>分析工具</h4>

<table>
<thead>
<tr>
<th></th>
<th> 模块  </th>
<th> 工具 </th>
<th>  参数  </th>
</tr>
</thead>
<tbody>
<tr>
<td></td>
<td>sequencing</td>
<td> HiSeq2000</td>
<td> PE150|</td>
</tr>
<tr>
<td></td>
<td>trimming   CLC Genomics </td>
<td>Workbench </td>
<td>MinQ=20,MinLen=50,adapter, no ambiguous nucleotides|</td>
</tr>
<tr>
<td></td>
<td>assembly   </td>
<td>CLC Genomics Workbench</td>
<td>    k-mer=63|</td>
</tr>
<tr>
<td></td>
<td>mapping</td>
<td>   CLC Genomics Workbench  </td>
<td>相似度>95%，覆盖度=100%|</td>
</tr>
<tr>
<td></td>
<td>PCA    </td>
<td>R  </td>
<td> |</td>
</tr>
<tr>
<td></td>
<td>ORF predict</td>
<td>   HMMER3  -cut_tc</td>
<td></td>
</tr>
<tr>
<td></td>
<td>align  </td>
<td>blastp，MEGAN </td>
<td>RefSeq protein database</td>
</tr>
<tr>
<td></td>
<td>plot   R   </td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td>visualization  </td>
<td>Cytoscape  </td>
<td></td>
</tr>
<tr>
<td></td>
<td>Re-assembly    </td>
<td>CLC &amp; Velvet   </td>
<td>k-mer=99</td>
</tr>
</tbody>
</table>


<h4>CLC Genomics Workbench</h4>

<p>由丹麦生物信息学软件公司CLC Bio（QIAGEN收购）开发<br/>
功能：分析各代测序数据—质控、组装、mapping、变异检测、注释、比较基因组学分析、图形化展示、差异表达分析、RNA二级结构预测、引物和探针设计、系统进化分析……<br/>
优点：图形化界面，多系统支持，自有算法，功能整合<br/>
缺点：贵</p>

<h4>其他binning工具</h4>

<p>TETRA：四核苷酸多态性的统计聚类<br/>
Phylopythia：基于支持向量机，通过已知序列的信息来判断宏基因组中某片段归为哪一类<br/>
SOrt-ITEMS：基于序列比对，鉴定分类地位，直系同源聚类<br/>
INDUS &amp; TWARIT：基于寡核苷酸的组成</p>

<h4>参考资料</h4>

<p>[1]<a href="http://blog.sina.com.cn/s/blog_670445240101lght.html">每日一生信—宏基因组聚类（binning\classification）</a><br/>
[2] Mande, Sharmila S.; Monzoorul Haque Mohammed; Tarini Shankar Ghosh (2012). &ldquo;Classification of metagenomic sequences: methods and challenges.&rdquo;. Briefings in Bioinformatics 13 (6): 669–81.<br/>
[3]<a href="http://omictools.com/binning-shotgun-reads-c191-p1.html">OMIStools</a></p>
</div>


  <footer>
    <p class="meta">
      
  

<span class="byline author vcard">Posted by <span class="fn">Wu Xiaomeng(Rachel)</span></span>

      








  


<time datetime="2014-09-18T12:06:07+08:00" pubdate data-updated="true">Sep 18<span>th</span>, 2014</time>
      

<span class="categories">
  
    <a class='category' href='/blog/category/wen-xian-fen-xiang-metagenome-binning/'>文献分享,metagenome,binning</a>
  
</span>


    </p>
    
      <div class="sharing">
  
  
	<!-- JiaThis Button BEGIN -->
<div class="jiathis_style_32x32">
<a class="jiathis_button_qzone"></a>
<a class="jiathis_button_tsina"></a>
<a class="jiathis_button_tqq"></a>
<a class="jiathis_button_weixin"></a>
<a class="jiathis_button_renren"></a>
<a href="http://www.jiathis.com/share?uid=1880878" class="jiathis jiathis_txt jtico jtico_jiathis" target="_blank"></a>
<a class="jiathis_counter_style"></a>
</div>
<script type="text/javascript">
var jiathis_config = {data_track_clickback:'true'};
</script>
<script type="text/javascript" src="http://v3.jiathis.com/code/jia.js?uid=1880878" charset="utf-8"></script>
<!-- JiaThis Button END -->

<!-- UY BEGIN -->
<div id="uyan_frame"></div>
<script type="text/javascript" src="http://v2.uyan.cc/code/uyan.js?uid=1880878"></script>
<!-- UY END -->

  
</div>

    
    <p class="meta">
      
        <a class="basic-alignment left articlenav" href="/blog/2014/09/12/yong-perlhua-svgtu/" title="Previous Post: 用perl画SVG图">&laquo; 用perl画SVG图</a>
      
      
    </p>
  </footer>
</article>

</div>

<aside class="sidebar">
  
    <section>
  <h1>Recent Posts</h1>
  <ul id="recent_posts">
    
      <li class="post">
        <a href="/blog/2014/09/18/wen-xian-fen-xiang-genome-sequences-of-rare/">文献分享-metagenome Binning</a>
      </li>
    
      <li class="post">
        <a href="/blog/2014/09/12/yong-perlhua-svgtu/">用perl画SVG图</a>
      </li>
    
      <li class="post">
        <a href="/blog/2014/07/16/deployshi-non-fast-forward-bao-cuo/">Deploy时[non-fast-forward]报错</a>
      </li>
    
      <li class="post">
        <a href="/blog/2014/04/21/psim-a-pilot-study-sequencing-simulator/">PSim-a Pilot Study Sequencing Simulator</a>
      </li>
    
      <li class="post">
        <a href="/blog/2014/04/17/podxie-fa/">Pod写法</a>
      </li>
    
  </ul>
</section>

<section>
  <h1>GitHub Repos</h1>
  <ul id="gh_repos">
    <li class="loading">Status updating...</li>
  </ul>
  
  <a href="https://github.com/rachelwxm">@rachelwxm</a> on GitHub
  
  <script type="text/javascript">
    $.domReady(function(){
        if (!window.jXHR){
            var jxhr = document.createElement('script');
            jxhr.type = 'text/javascript';
            jxhr.src = '/javascripts/libs/jXHR.js';
            var s = document.getElementsByTagName('script')[0];
            s.parentNode.insertBefore(jxhr, s);
        }

        github.showRepos({
            user: 'rachelwxm',
            count: 0,
            skip_forks: true,
            target: '#gh_repos'
        });
    });
  </script>
  <script src="/javascripts/github.js" type="text/javascript"> </script>
</section>

<section>
<h1>Categories</h1>
<ul id="categories">
<li class='category'><a href='/blog/category/computing-for-data-analysis/'>Computing_for_Data_Analysis (7)</a></li>
<li class='category'><a href='/blog/category/malbac/'>MALBAC (1)</a></li>
<li class='category'><a href='/blog/category/mda/'>MDA (1)</a></li>
<li class='category'><a href='/blog/category/psim/'>PSim (1)</a></li>
<li class='category'><a href='/blog/category/r/'>R (7)</a></li>
<li class='category'><a href='/blog/category/github/'>github (1)</a></li>
<li class='category'><a href='/blog/category/octopress/'>octopress (3)</a></li>
<li class='category'><a href='/blog/category/octopress-/'>octopress, (1)</a></li>
<li class='category'><a href='/blog/category/perl/'>perl (1)</a></li>
<li class='category'><a href='/blog/category/perl-/'>perl, (1)</a></li>
<li class='category'><a href='/blog/category/pod/'>pod (1)</a></li>
<li class='category'><a href='/blog/category/svg-/'>svg, (1)</a></li>
<li class='category'><a href='/blog/category/文献分享-metagenome-binning/'>文献分享,metagenome,binning (1)</a></li>
<li class='category'><a href='/blog/category/画图/'>画图 (1)</a></li>
 
</ul>
</section>
<section>
  <h1>友情链接</h1>
  <a href="http://www.52nlp.cn/" target="_blank">我爱自然语言处理</a>
  <br>
  <a href="http://www.cnblogs.com/tornadomeet/" target="_blank">tornadomeet-一个很牛掰的博客</a>
  <br>
  <a href="https://scowiki.wordpress.com/" target="_blank">BENM</a>
  <br>
  <a href="http://www.plob.org/" target="_blank">Public Library of Bioinformatics</a>
  <br>
  <a href="http://bioops.info/" target="_blank">Bioops</a>
  <br>
  <a href="http://bioinfo.cs.ccu.edu.tw/wiki/doku.php" target="_blank">生物資訊教學-台湾</a>
  <br>
  
</section>
<section>
<h1>最新评论</h1>
<div id="uyan_newcmt_unit"></div>
<script type="text/javascript" src="http://v2.uyan.cc/code/uyan.js?uid=1850190"></script>
</section>

  
</aside>


    </div>
  </div>
  <footer role="contentinfo"><p>
  Copyright &copy; 2014 - Wu Xiaomeng(Rachel) -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>
  <script type="text/javascript">var cnzz_protocol = (("https:" == document.location.protocol) ? " https://" : " http://");document.write(unescape("%3Cspan id='cnzz_stat_icon_1000378192'%3E%3C/span%3E%3Cscript src='" + cnzz_protocol + "v1.cnzz.com/z_stat.php%3Fid%3D1000378192' type='text/javascript'%3E%3C/script%3E"));</script>
</p>

</footer>
  











</body>
</html>
